converter.py

```diff
@@ -0,0 +1,183 @@
+from abc import ABC, abstractmethod, abstractproperty
+from functools import reduce
+from dataclasses import dataclass, field
+from os import path
+
+
+@dataclass
+class Geometry(ABC):
+    pass
+
+
+@dataclass(order=True, frozen=True)
+class Cylinder(Geometry):
+    sort_index: int = field(init=False)
+
+    id: str
+    radius: int = 10
+    height: int = 10
+
+    template: str = """Geometry Cyl
+    Name ScoringCylinder
+    R 0.0 10.0 {cyl_nr:d}
+    Z 0.0 30.0 {cyl_nz:d}"""
+
+    def __post_init__(self):
+        object.__setattr__(self, 'sort_index', self.id)
+
+    def __str__(self) -> str:
+        return self.template.format(cyl_nr=self.radius, cyl_nz=self.height)
+
+
+@dataclass(order=True, frozen=True)
+class Mesh(Geometry):
+    sort_index: int = field(init=False)
+
+    id: str
+    x: int = 10
+    y: int = 10
+    z: int = 10
+
+    template: str = """Geometry Mesh
+    Name MyMesh_YZ
+    X -5.0  5.0    {mesh_nx:d}
+    Y -5.0  5.0    {mesh_ny:d}
+    Z  0.0  30.0   {mesh_nz:d}"""
+
+    def __post_init__(self):
+        object.__setattr__(self, 'sort_index', self.id)
+
+    def __str__(self) -> str:
+        return self.template.format(mesh_nx=self.x, mesh_ny=self.y, mesh_nz=self.z)
+
+
+class BeamConfig:
```

Comment:
use dataclass as well

Reply:
Resolved in 3745118fadacd3eefeaf2cee5b591052a549786f